src/storage/src/hummock/conflict_detector.rs

```diff
@@ -99,9 +105,27 @@ impl ConflictDetector {
     }
 
     /// Archives an epoch. An archived epoch cannot be written anymore.
-    pub fn archive_epoch(&self, epoch: HummockEpoch) {
-        self.epoch_history.remove(&epoch);
-        self.set_watermark(epoch);
+    pub fn archive_epoch(&self, epoch: HummockEpoch, first_epoch: Option<HummockEpoch>) {
+        assert!(
+            epoch > self.get_epoch_watermark(),
+            "write to an archived epoch: {} , c_epoch :{}",
+            epoch,
+            self.get_epoch_watermark(),
+        );
+        if self.epoch_history.contains_key(&epoch) {
+            assert!(
+                self.epoch_history.get(&epoch).is_some(),
+                "epoch has been archived: epoch is {}",
+                epoch
+            );
+        }
+        self.epoch_history.insert(epoch, None);
+        if let Some(first_epoch) = first_epoch {
+            if first_epoch - 1 != self.get_epoch_watermark() {
```

Comment:
It seems that `first_epoch` is the minimum epoch in shared buffer. But I think the `first_epoch` is not increasing monotonically? 

For example let's assume that we have two actor with different current epochs `e` and `e+1`.  It is possible the actor with `e+1` writes before the actor with `e`. Before the actor with `e` writes, the `first_epoch` is `e+1`, but after it writes, the `first_epoch` become `e`.

Reply:
ohhh, I find the minimum epoch in shared buffer is ` new_pinned_version.max_committed_epoch` , Is it not increasing monotonically??

Reply:
The `get_first_epoch` in `LocalVersion` simply return the minimum epoch in the `BTreeMap`. As in the example above, seems that the returned epoch is not necessarily `new_pinned_version.max_committed_epoch`.

Reply:
shall we use `new_pinned_version.max_committed_epoch` directly then?

Reply:
I want to use `pinned_version.max_committed_epoch`, because I do not think we need to write an epoch large than it.

Reply:
What do you mean by `first_epoch - 1`? The generated epoch number is monotonically increasing, however it doesn't have to be continuous.

Reply:
yes, In this,  I do not remove the first_epoch in  epoch_history, because it may be written

Reply:
I suggest keeping the original `archive_epoch` interface without introducing `first_epoch` and add another interface (e.g. `commit_epoch`) invoked on `LocalVersion::set_pinned_version` to delete epoch history and bump epoch watermark.

Reply:
okok , i wil do it